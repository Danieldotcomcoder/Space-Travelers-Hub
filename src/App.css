:root {
  --main-color:  #00A3B8;
  --text-color: #fff;
}

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  list-style: none;
}

nav {
  display: flex;
  flex-direction: row;
}

.logo {
  display: flex;
  flex-direction: row;
}

h1 {
  margin-left: 10px;
  margin-top: 20px;
}

.logo-img {
  width: 50px;
  margin-left: 26px;
  margin-top: 10px;
}

.nav-link {
  display: flex;
  flex-direction: row;
  margin-left: 950px;
  margin-top: 22px;
}

.nav-link a {
  text-decoration: none;
  margin: 22px;
  padding: 10px;
  font-size: 18px;
  color: var(--main-color);
  cursor: pointer;
}

.rocket-c {
  display: flex;
  flex-direction: column;
  margin: 22px;
}

.rocket-list {
  display: flex;
  flex-direction: row;
  margin: 12px;
}

.rocket-img {
  width: 160px;
  margin-right: 20px;
}

.rocket-info {
  display: flex;
  flex-direction: column;
}

.rocket-des {
  font-size: 16px;
  margin: 10px 0;
}

.reserve-btn {
  max-width: 120px;
  background-color: var(--main-color);
  border: none;
  border-radius: 3px;
  color: #fff;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 12px;
  margin: 4px 2px;
  padding: 4px 2px;
  cursor: pointer;
}

.reserved-r {
  color: #fff;
  background-color: var(--main-color);
  border-radius: 10px;
  margin: 0 8px 0 0;
  padding: 3px 8px;
  font-size: 12px;
}

.cancel-btn {
  max-width: 120px;
  background-color:  #e0e0e0;
  border: 1px solid #cfcfcf;
  border-radius: 3px;
  color: #918e8e;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 12px;
  margin: 4px 2px;
  padding: 4px 2px;
  cursor: pointer;
}

.my-rockets {
  margin: 10px;
}

h2{
  margin: 10px;
}

.rocket-profile {
  display: flex;
  flex-direction: column;
  max-width: 40vw;
  border: 1px solid #918e8e;
}

.rocket-name {
  margin: 0;
  padding: 12px;
  font-size: 20px;
  border-bottom: 1px solid #918e8e;
}
