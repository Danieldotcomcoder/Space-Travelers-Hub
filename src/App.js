import {
  BrowserRouter as Router, Route, Routes, NavLink,
} from 'react-router-dom';
import Rocket from './Components/RocketPage';
import Mission from './Components/MissionPage';
import MyProfile from './Components/MyProfile';
import logo from './images/logo.png';
import './App.css';

function App() {
  return (
    <div className="App">
      <Router>
        <div>
          <nav>
            <div className="logo">
              <img className="logo-img" src={logo} alt="logo" />
              <h1>Space Travelers&apos; Hub</h1>
            </div>
            <ul className="nav-link">
              <li>
                <NavLink exact to="/">Rocket</NavLink>
              </li>
              <li>
                <NavLink to="/Mission">Mission</NavLink>
              </li>
              <li>
                <NavLink to="/MyProfile"> My Profile</NavLink>
              </li>
            </ul>
          </nav>
          <Routes>
            <Route path="/" element={<Rocket />} />
            <Route path="/Mission" element={<Mission />} />
            <Route path="/MyProfile" element={<MyProfile />} />
          </Routes>
        </div>
      </Router>
    </div>
  );
}

export default App;
