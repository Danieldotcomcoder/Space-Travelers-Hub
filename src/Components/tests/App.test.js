import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import App from '../../App';
import store from '../../Redux/configureStore';

const APP = (
  <Provider store={store}>
    <App />
  </Provider>
);

describe('App test', () => {
  it('Render App', () => {
    const app = renderer.create(APP).toJSON();
    expect(app).toMatchSnapshot();
  });
});
