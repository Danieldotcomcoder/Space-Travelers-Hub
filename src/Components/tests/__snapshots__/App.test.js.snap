// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test Render App 1`] = `
<div
  className="App"
>
  <div>
    <nav>
      <div
        className="logo"
      >
        <img
          alt="logo"
          className="logo-img"
          src="logo.png"
        />
        <h1>
          Space Travelers' Hub
        </h1>
      </div>
      <ul
        className="nav-link"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Rocket
          </a>
        </li>
        <li>
          <a
            className=""
            href="/Mission"
            onClick={[Function]}
          >
            Mission
          </a>
        </li>
        <li>
          <a
            className=""
            href="/MyProfile"
            onClick={[Function]}
          >
             My Profile
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="rocket-c"
    />
  </div>
</div>
`;
